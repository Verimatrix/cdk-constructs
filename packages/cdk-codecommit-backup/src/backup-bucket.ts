import { Duration, RemovalPolicy } from 'aws-cdk-lib';
import { BlockPublicAccess, Bucket, BucketEncryption, BucketProps } from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

export interface BackupBucketProps extends BucketProps {
  readonly retentionPeriod?: Duration;
}

export class BackupBucket extends Bucket {
  constructor(scope: Construct, id: string, props?: BackupBucketProps) {
    const { retentionPeriod = Duration.days(35), ...rest } = props || {};

    super(scope, id, {
      lifecycleRules: [
        {
          expiration: retentionPeriod,
        },
      ],
      encryption: BucketEncryption.S3_MANAGED,
      publicReadAccess: false,
      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
      removalPolicy: RemovalPolicy.RETAIN,
      ...rest,
    });
  }
}
