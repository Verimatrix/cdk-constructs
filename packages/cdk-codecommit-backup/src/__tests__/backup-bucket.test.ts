import 'jest-cdk-snapshot';
import { Duration, Stack } from 'aws-cdk-lib';
import { BackupBucket } from '../backup-bucket';

test('default setup', (): void => {
  const stack = new Stack();

  new BackupBucket(stack, 'BackupBucket');

  expect(stack).toMatchCdkSnapshot();
});

test('retentionPeriod', (): void => {
  const stack = new Stack();

  new BackupBucket(stack, 'BackupBucket', {
    retentionPeriod: Duration.days(4711),
  });

  expect(stack).toMatchCdkSnapshot();
});
