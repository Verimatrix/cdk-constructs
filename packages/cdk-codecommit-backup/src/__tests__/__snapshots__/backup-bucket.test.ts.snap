// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default setup 1`] = `
Object {
  "Resources": Object {
    "BackupBucket26B8E51C": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LifecycleConfiguration": Object {
          "Rules": Array [
            Object {
              "ExpirationInDays": 35,
              "Status": "Enabled",
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;

exports[`retentionPeriod 1`] = `
Object {
  "Resources": Object {
    "BackupBucket26B8E51C": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "LifecycleConfiguration": Object {
          "Rules": Array [
            Object {
              "ExpirationInDays": 4711,
              "Status": "Enabled",
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
